### PR TITLE
docs(Backend): add basic endpoints

### LINKED ISSUE #2: Document API
Create README.md file under backend folder with a table describing all endpoints that the api should have

### PR DESCRIPTION
**Pull request by jfmrm**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,8 @@
 | Route | Description | Parameters | Expected Output |
 |-------|-------------|------------|-----------------|
 | `POST /api/login`| logs the user in | body: `{ email: string, password: string}` | `{ token: string, refresh_token: string }` |
+| `POST /api/user?=1/create`| Create a list | body `{ title: string, description: string}` | `{ title: string, description: string}` |
```
Comment (jfmrm) [inline README.md]: this path is wrong, it wont work. You should take a look at REST pattern. This path would be: `POST /api/user/{id}`
where {id} is a parameter

File: README.md [hunk 2]
```diff
@@ -3,3 +3,8 @@
...
+| `POST /api/sign`| Sign up user in | body: `{ firstName: string, lastName: string, email: string, password: string}` | `{ token: string, refresh_token: string }` |
```
Comment (jfmrm) [inline README.md]: maybe change `/api/sign` to `/api/sign-in`

File: README.md [hunk 3]
```diff
@@ -3,3 +3,8 @@
...
+| `GET /api/user?=1/review`| Review task | body: | `{ title: string, description: string}`|
+| `PUT /api/user?=1/review/edit`| Edit Task | body: `{ title: string, description: string}` | `{ title: string, description: string}` |
```
Comment (jfmrm) [inline README.md]: these paths seems to me that are related to tasks. Task is another entity in our system so it should be something like this (if you are following the REST pattern):
`PATCH /api/task/{id}/review`
`PATCH /api/task/{id}/edit`

File: README.md (hunk 1 above)
Comment (valter-junior) [reply README.md]: | `POST /api/user/task` | Create a List | body `{ title: string, description: string}` | `{ title: string, description: string}` |

File: README.md (hunk 2 above)
Comment (valter-junior) [reply README.md]: |`POST /api/sign-in` | Sign in a user | body `{ firstName: string, lastName: string, email: string, password: string}` | `{ token: string, refresh_token: string}`|

File: README.md (hunk 3 above)
Comment (valter-junior) [reply README.md]: | `GET /api/user/task` | List all task | body `{title: string, description: string}` | `{title: string, description: string}` |
| `GET /api/user/task/{id}` | List specific task | body `{title: string, description: string}` | `{title: string, description: string}` |
| `PUT /api/user/task/{id}/edit` | Edit specific task | body `{title: string, description: string}` | `{title: string, description: string}` |
| `PUT /api/user/task/{id}/done` | Realized specific task | body `{title: string, description: string}` | `{title: string, description: string}` |
| `DELETE /api/user/task/{id}/delete` | Delete specific task | body `{title: string, description: string}` | `{title: null, description: null}` |

File: README.md [hunk 4]
```diff
@@ -3,3 +3,10 @@
...
 |-------|-------------|------------|-----------------|
 | `POST /api/login`| logs the user in | body: `{ email: string, password: string}` | `{ token: string, refresh_token: string }` |
+| `POST /api/sign-in` | Sign in a user | body `{ firstName: string, lastName: string, email: string, password: string}` | `{ token: string, refresh_token: string}`|
+| `POST /api/user/task` | Create a List | body `{ title: string, description: string}` | `{ title: string, description: string}` |
+| `GET /api/user/task` | List all task | body `{title: string, description: string}` | `{title: string, description: string}` |
+| `GET /api/user/task/{id}` | List specific task | body `{title: string, description: string}` | `{title: string, description: string}` |
+| `PUT /api/user/task/{id}/edit` | Edit specific task | body `{title: string, description: string}` | `{title: string, description: string}` |
+| `PUT /api/user/task/{id}/done` | Realized specific task | body `{title: string, description: string}` | `{title: string, description: string}` |
+| `DELETE /api/user/task/{id}/delete` | Delete specific task | body `{title: string, description: string}` | `{title: null, description: null}` |
```
Comment (jfmrm) [inline README.md]: you don't need the last part of these paths.
PUT already implies that you are editing, and you don't need 2 paths to edit the same entity, you use the same path changing only the body content.

Also DELETE http method already implies deletion

Comment (valter-junior) [reply README.md]: Done

**Comment by jfmrm** [comment]
The paths are all cool, but I have a question.
On the backend, how are you planning to check which user is making the request, and if that user can actually do that request (maybe someone sends a request deleting a task that is not his own)?

**Review by valter-junior — COMMENTED** [review]
> The paths are all cool, but I have a question.
> On the backend, how are you planning to check which user is making the request, and if that user can actually do that request (maybe someone sends a request deleting a task that is not his own)?

Can you explain better how I do it?

**Comment by valter-junior** [comment]
> The paths are all cool, but I have a question.
> On the backend, how are you planning to check which user is making the request, and if that user can actually do that request (maybe someone sends a request deleting a task that is not his own)?

**Comment by valter-junior** [comment]
> > The paths are all cool, but I have a question.
> > On the backend, how are you planning to check which user is making the request, and if that user can actually do that request (maybe someone sends a request deleting a task that is not his own)?
> 
> Can you explain better how I do it?

Ok, Can I use OAuth 2.0? For on the backend check if the user can do the request and verify if a task deleting is his own?

**Pull request merged**